### PR TITLE
ROX-26594: Implement GetComplianceReport API

### PR DESCRIPTION
**Pull request by lvalerom**
### Description

Implement the API endpoints to retrieve the report history.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [x] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

- [x] Unit tests added
- [x] Manual tested the endpoints

### CODE REVIEW

File: central/complianceoperator/v2/scanconfigurations/service/service_impl_test.go [hunk 1]
```diff
@@ -656,6 +661,190 @@ func (s *ComplianceScanConfigServiceTestSuite) TestRunReport() {
 	s.Equal(v2.ComplianceRunReportResponse_SUBMITTED, resp.RunState, "Failed to submit report")
 }
 
+func (s *ComplianceScanConfigServiceTestSuite) TestGetReportHistory() {
+	s.T().Setenv(features.ComplianceReporting.EnvVar(), "true")
+	if !features.ComplianceReporting.Enabled() {
+		s.T().Skipf("Skip test when the compliance reporting feature flag is disabled")
```
Comment (vikin91) [inline central/complianceoperator/v2/scanconfigurations/service/service_impl_test.go:667]: Style: I think, that should be sufficient
```suggestion
		s.T().Skipf("compliance reporting feature flag is disabled")
```

**Review by vikin91 — COMMENTED** [review]
Looks good! 
Could we only test or check that `nil, nil` situation in `convertStorageSnapshotsToV2Snapshots` (see comment)?

File: central/complianceoperator/v2/scanconfigurations/service/convert.go [hunk 2]
```diff
@@ -358,3 +376,57 @@ func convertStorageScanConfigToV2ScanStatuses(ctx context.Context,
...
+	if snapshots == nil {
+		return nil, nil
+	}
+	retSnapshots := make([]*v2.ComplianceReportSnapshot, 0, len(snapshots))
+	for _, snapshot := range snapshots {
+		converted, err := convertStorageSnapshotToV2Snapshot(ctx, snapshot, configDS, bindingDS, suiteDS, notifierDS)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to convert storage compliance operator report snapshot %s to response", snapshot.GetName())
+		}
+		retSnapshots = append(retSnapshots, converted)
```
[the remark this answers is not in the document]
Comment (lvalerom) [reply central/complianceoperator/v2/scanconfigurations/service/convert.go:394]: I added an if to skip the snapshot (and a unit test) if we land in this situation but I think it is impossible (?) to land in the `return nil nil` as the snapshot slice is returned from the db and I'd expect those snapshots to not be nil.

File: central/complianceoperator/v2/scanconfigurations/service/service_impl_test.go (hunk 1 above)
Comment (lvalerom) [reply central/complianceoperator/v2/scanconfigurations/service/service_impl_test.go]: Done

**Review by vikin91 — APPROVED** [review]
Thanks for improving, that looks good!

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**